vm/src/obj/objint.rs

```diff
@@ -792,13 +792,19 @@ fn str_to_int(vm: &VirtualMachine, literal: &str, base: &BigInt) -> PyResult<Big
     // try to find base
     if let Some(radix_candidate) = radix_candidate {
         if let Some(matched_radix) = detect_base(&radix_candidate) {
-            if base_u32 != 0 && base_u32 != matched_radix {
-                return Err(invalid_literal(vm, literal, base));
-            } else {
+            if base_u32 == 0 || base_u32 == matched_radix {
                 base_u32 = matched_radix;
-            }
+                buf.drain(radix_range);
 
-            buf.drain(radix_range);
+                if buf.starts_with('_') {
+                    buf.remove(0);
+                }
+            } else if (matched_radix == 2 && base_u32 < 12)
```

Comment:
I tried to understand this code, but how does it work? Could you add a small comment to explain what is happening here? Do you check ascii codes?

Reply:
This is very simple. It is code to fix `'0x_1'` is unvalidate (issue : #1420)
First, we need to know that `'_1'` is invalid literal (Also it is filtered `validate_literal()`)
 
In current logic, After we detect validate radix, then remove the radix from literal. 
So if executed `int('0x_1', base=16)` and after doing `buf.drain(radix_range)`, literal will be considered `_1`. As we saw above it is invalid literal.

So i just remove the first underscore when `buf.drain(radix_range)` exectued, So literal will be considered `1`. I select this method Because it is easy to implement and do not hurt current logic.